<launch>
  <!-- Map file path -->
  <arg name="map_path" default="$(find mall_navigation)/maps"/>
  <arg name="semantic_map_file" default="DISB1.yaml"/>
  
  <!-- Start the MarRTino navigation stack -->
  <include file="$(env MARRTINO_APPS_HOME)/navigation/launch/amcl.launch">
    <arg name="map_file" value="$(arg map_path)/dis-B1-2011-09-27.yaml"/>
  </include>
  
  <include file="$(env MARRTINO_APPS_HOME)/navigation/launch/move_base.launch"/>
  
  <!-- Start the Mall Navigation system -->
  <node pkg="mall_navigation" type="mall_navigation_node.py" name="mall_navigator" output="screen">
    <param name="map_path" value="$(arg map_path)"/>
    <param name="semantic_map_file" value="$(arg semantic_map_file)"/>
  </node>
  
  <!-- Start the status listener -->
  <node pkg="mall_navigation" type="status_listener.py" name="status_listener" output="screen"/>
  
  <!-- Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mall_navigation)/rviz/mall_navigation.rviz" if="$(arg use_rviz)"/>
</launch>